compiler/rustc_builtin_macros/src/deriving/generic/mod.rs

```diff
@@ -995,12 +995,14 @@ impl<'a> MethodDef<'a> {
     /// `&self.x` because that might cause an unaligned ref. So for any trait
     /// method that takes a reference, if the struct impls `Copy` then we use a
     /// local block to force a copy:
-    /// ```
+    /// ```rust
+    /// use std::hash::Hash;
+    ///
     /// # struct A { x: u8, y: u8 }
     /// impl PartialEq for A {
     ///     fn eq(&self, other: &A) -> bool {
     ///         // Desugars to `{ self.x }.eq(&{ other.y }) && ...`
-    ///         { self.x } == { other.y } && { self.y } == { other.y }
+    ///         self.x == { other.y } && { self.y } == { other.y }
```

Comment:
It actually expands to the left code though.
```rust
#[derive(PartialEq, Clone, Copy)]
#[repr(packed)]
struct A { x: u8, y: u8 }
```
```rust
impl ::core::cmp::PartialEq for A {
    #[inline]
    fn eq(&self, other: &A) -> bool {
        { self.x } == { other.x } && { self.y } == { other.y }
    }
}
```
So you should `ignore` it instead I guess